File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/knowledge_base.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import DocumentAddRequest, DocumentResponse, KnowledgeBaseCreateRequest, KnowledgeBaseListResponse, KnowledgeBaseResponse, KnowledgeSearchRequest, KnowledgeSearchResponse, ListDocumentsGetResponse

class KnowledgeBaseApi:
    """knowledge_base API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def list_documents(self, knowledge_base_id: str, params: Optional[Dict[str, Any]] = None) -> ListDocumentsGetResponse:
        """List documents"""
        return self._client.get(f"/ai/v3/knowledge-bases/{knowledge_base_id}/documents", params=params)

    def add_document(self, knowledge_base_id: str, body: DocumentAddRequest) -> DocumentResponse:
        """Add document"""
        return self._client.post(f"/ai/v3/knowledge-bases/{knowledge_base_id}/documents", json=body)

    def search_knowledge(self, body: KnowledgeSearchRequest) -> KnowledgeSearchResponse:
        """Search knowledge"""
        return self._client.post(f"/ai/v3/knowledge-bases/search", json=body)

    def list_knowledge_bases(self, params: Optional[Dict[str, Any]] = None) -> KnowledgeBaseListResponse:
        """List knowledge bases"""
        return self._client.get(f"/ai/v3/knowledge-bases", params=params)

    def create_knowledge_base(self, body: KnowledgeBaseCreateRequest) -> KnowledgeBaseResponse:
        """Create knowledge base"""
        return self._client.post(f"/ai/v3/knowledge-bases", json=body)

    def get_document(self, knowledge_base_id: str, document_id: str) -> DocumentResponse:
        """Get document"""
        return self._client.get(f"/ai/v3/knowledge-bases/{knowledge_base_id}/documents/{document_id}")

    def delete_document(self, knowledge_base_id: str, document_id: str) -> None:
        """Delete document"""
        return self._client.delete(f"/ai/v3/knowledge-bases/{knowledge_base_id}/documents/{document_id}")

    def get_knowledge_base(self, knowledge_base_id: str) -> KnowledgeBaseResponse:
        """Get knowledge base"""
        return self._client.get(f"/ai/v3/knowledge-bases/{knowledge_base_id}")

    def delete_knowledge_base(self, knowledge_base_id: str) -> None:
        """Delete knowledge base"""
        return self._client.delete(f"/ai/v3/knowledge-bases/{knowledge_base_id}")

File: sdkwork-ai-sdk-flutter/lib/src/api/knowledge_base.dart
import '../http/client.dart';
import '../models.dart';

class KnowledgeBaseApi {
  final HttpClient _client;
  
  KnowledgeBaseApi(this._client);

  /// List documents
  Future<ListDocumentsGetResponse?> listDocuments(String knowledge_base_id, Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.aiPath('/knowledge-bases/${knowledge_base_id}/documents'), params: params);
    return response is ListDocumentsGetResponse ? response : null;
  }

  /// Add document
  Future<DocumentResponse?> addDocument(String knowledge_base_id, DocumentAddRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/knowledge-bases/${knowledge_base_id}/documents'), body: body, contentType: 'application/json');
    return response is DocumentResponse ? response : null;
  }

  /// Search knowledge
  Future<KnowledgeSearchResponse?> searchKnowledge(KnowledgeSearchRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/knowledge-bases/search'), body: body, contentType: 'application/json');
    return response is KnowledgeSearchResponse ? response : null;
  }

  /// List knowledge bases
  Future<KnowledgeBaseListResponse?> listKnowledgeBases(Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.aiPath('/knowledge-bases'), params: params);
    return response is KnowledgeBaseListResponse ? response : null;
  }

  /// Create knowledge base
  Future<KnowledgeBaseResponse?> createKnowledgeBase(KnowledgeBaseCreateRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/knowledge-bases'), body: body, contentType: 'application/json');
    return response is KnowledgeBaseResponse ? response : null;
  }

  /// Get document
  Future<DocumentResponse?> getDocument(String knowledge_base_id, String document_id) async {
    final response = await _client.get(ApiPaths.aiPath('/knowledge-bases/${knowledge_base_id}/documents/${document_id}'));
    return response is DocumentResponse ? response : null;
  }

  /// Delete document
  Future<void> deleteDocument(String knowledge_base_id, String document_id) async {
    await _client.delete(ApiPaths.aiPath('/knowledge-bases/${knowledge_base_id}/documents/${document_id}'));
  }

  /// Get knowledge base
  Future<KnowledgeBaseResponse?> getKnowledgeBase(String knowledge_base_id) async {
    final response = await _client.get(ApiPaths.aiPath('/knowledge-bases/${knowledge_base_id}'));
    return response is KnowledgeBaseResponse ? response : null;
  }

  /// Delete knowledge base
  Future<void> deleteKnowledgeBase(String knowledge_base_id) async {
    await _client.delete(ApiPaths.aiPath('/knowledge-bases/${knowledge_base_id}'));
  }
}
